### ISSUE TITLE
Add an AreaRectContainer node

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#8276]
### Describe the project you are working on

A spaceship game

### Describe the problem or limitation you are having in your project

I need to autosize an area2d inside a control
![image](https://github.com/godotengine/godot-proposals/assets/2453707/6dc5f315-74b4-4f70-ba70-25e6c7a1c7dd)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a AreaRectContainer which would detect it's area2d's and would be able to use the gui layout system. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

AreaRectContainer would be like a area2d but instead of extending from node2d it would extend from container and would only allow rectangleshape2d's

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It would be used when making multiplayer cursor based gui.

### Is there a reason why this should be core and not an add-on in the asset library?

There really isn't a good solution for something like this yet.

**Comment by rsubtil** [comment]
This would imply mixing UI nodes with 2D nodes, which honestly sounds like it would create a lot of trouble.

Besides, you can work around this with a script by detecting size changes (`resized` signal), getting the control's rect (`get_global_rect`), and then moving and resizing your area 2D accordingly.

**Comment by Shadowblitz16** [comment]
UI and 2D logic are mixed all the time in games.
The idea is that we would be able to easily implement collision logic with 2d elements.

2D virtual cursors are just an example of somethings they can be used for.
2D and UI collision is used all the time in games.

anything where the user wants to either separate the cursor from the mouse and/or add collisions with the world while working with ui

**Comment by rsubtil** [comment]
The issue is that since Godot chose to separate UI and 2D at a fundamental level (to the point where setting position and size on `Control` and `Node2D` is very different), if you really need to use both, you'll need to handle such intricacies according to what your game needs. And it's not an universally applicable solution.

Your proposal is extremely specific to your needs, and not everyone is using UI and 2D collision mixed in in their games like you do. Others might be using different 2D physics objects, or just plain `Node2D`s. Or maybe have come up with solutions that can be done exclusively through `Control` or `Node2D` types. And again, if it can "be worked around with a few lines of script", then it should be done that way.

**Comment by Shadowblitz16** [comment]
> The issue is that since Godot chose to separate UI and 2D at a fundamental level (to the point where setting position and size on `Control` and `Node2D` is very different), if you really need to use both, you'll need to handle such intricacies according to what your game needs. And it's not an universally applicable solution.
> 
> Your proposal is extremely specific to your needs, and not everyone is using UI and 2D collision mixed in in their games like you do. Others might be using different 2D physics objects, or just plain `Node2D`s. Or maybe have come up with solutions that can be done exclusively through `Control` or `Node2D` types. And again, if it can "be worked around with a few lines of script", then it should be done that way.

there is more then my purpose.

who gets to say it's only my specific use case?
you guys literally added a label3d because of this sort of thing.

why is there no built in engine way to cross interact with node2d node3d and controls?

what about raycasting to controls from 3d? 
I had to write my own grid2d because grid container didn't work with node2d's and I wanted a tilemap of nodes.
What if someone wanted to layout 3d nodes in a grid container based on perspective? 

You guys always assume what people need and since there is really no way to use base godot nodes to make any sort of game without attaching scripts to them shows the absolute horid reusability of the godot engine.

This is that exact reason why things like composable gameobjects and ecs are used.

**Comment by rsubtil** [comment]
> who gets to say it's only my specific use case?
you guys literally added a label3d because of this sort of thing.

Well yes, that's the point of these proposals, to propose something for Godot and gauge community interest and feedback to see if it's a good idea or not. Label3D went through the same process (https://github.com/godotengine/godot-proposals/issues/3219). I don't get to say if it makes in or not (especially because I'm not a Godot core dev), I'm just giving my opinion on your proposal.

> why is there no built in engine way to cross interact with node2d node3d and controls?

> what about raycasting to controls from 3d?

> This is that exact reason why things like composable gameobjects and ecs are used.

All I can say here is that every game engine is built differently, and this implies different workflows. Godot chose to do a very rigid separation here, and I'm not going to pretend it's perfect, but I can say my experience coming from other engines also conditioned me to try to mix these features initially, and I had a horrible time until I accepted that I needed to "forget" my knowledge and learn how to do things "the Godot way".
